ts/session/messages/outgoing/controlMessage/MessageRequestResponse.ts

```diff
@@ -1,18 +1,47 @@
+import ByteBuffer from 'bytebuffer';
+import { isEmpty } from 'lodash';
 import { SignalService } from '../../../../protobuf';
+import { LokiProfile } from '../../../../types/Message';
 import { ContentMessage } from '../ContentMessage';
 import { MessageParams } from '../Message';
 
 // tslint:disable-next-line: no-empty-interface
-export interface MessageRequestResponseParams extends MessageParams {}
+export interface MessageRequestResponseParams extends MessageParams {
+  lokiProfile?: LokiProfile;
+}
 
 export class MessageRequestResponse extends ContentMessage {
   // we actually send a response only if it is an accept
   // private readonly isApproved: boolean;
+  private readonly profileKey?: Uint8Array;
+  private readonly displayName?: string;
+  private readonly avatarPointer?: string;
 
   constructor(params: MessageRequestResponseParams) {
     super({
       timestamp: params.timestamp,
     } as MessageRequestResponseParams);
+
+    if (params.lokiProfile && params.lokiProfile.profileKey) {
+      if (
+        params.lokiProfile.profileKey instanceof Uint8Array ||
+        (params.lokiProfile.profileKey as any) instanceof ByteBuffer
+      ) {
+        this.profileKey = new Uint8Array(params.lokiProfile.profileKey);
+      } else {
+        this.profileKey = new Uint8Array(
+          ByteBuffer.wrap(params.lokiProfile.profileKey).toArrayBuffer()
+        );
+      }
+    }
+
+    this.displayName = params.lokiProfile && params.lokiProfile.displayName;
+
+    // no need to iclude the avatarPointer if there is no profileKey associated with it.
```

Comment:
This code seems to be duplicated inside of ts/session/messages/outgoing/visibleMessage/VisibleMessage.ts. Can we make it into it's own function / check?

Reply:
amended